src/main/kotlin/com/cosmotech/api/config/CsmPlatformProperties.kt

```diff
@@ -276,7 +276,17 @@ data class CsmPlatformProperties(
 
         /** Don't try to use node selector to schedule argo workflows */
         val ignoreNodeSelector: Boolean = false,
-    )
+
+        /** enable secret mount on workflow */
+        val mountSecret: Boolean = false,
+
+        /** Info on k8s secret to mount on workflow */
+        val secrets: List<ArgoSecrets> = emptyList()
+    ) {
+      data class ArgoSecrets(val name: String, val keyPath: List<ArgoKeyPath>) {
+        data class ArgoKeyPath(val key: String, val path: String)
+      }
+    }
```

Comment:
Could you please rename `Argo* `objects ?
It could be specified as `extraSecrets` and boolean `mountSecret` could be removed I guess. On API side, secrets could be mounted directly if `extraSecrets` is empty